### DISCUSSION TITLE
CRITICAL OSError(13) regarding output json file permission

### DISCUSSION THREAD

**Discussion by whatthejay** [prowler-cloud/prowler#1861] [Q&A] [answered]
Hi, Team.

I'm using 3.1.4 version of Prowler.

When I run prowler to audit my aws account, it shows error and exit.
I tested in two ubuntu machines and both are same.
When I specify output modes by "prowler aws --output-modes csv" then it works fine but I need to use JSON output to automate the whole audit process.

Can someone please help me to troubleshoot this?

===========================================================

(prowler) ubuntu@mkvmubuntu:~/prowler$ prowler aws
                         _
 _ __  _ __ _____      _| | ___ _ __
| '_ \| '__/ _ \ \ /\ / / |/ _ \ '__|
| |_) | | | (_) \ V  V /| |  __/ |
| .__/|_|  \___/ \_/\_/ |_|\___|_|v3.1.4
|_| the handy cloud security tool

Date: 2023-02-08 13:19:20

This report is being generated using credentials below:

AWS-CLI Profile: [default] AWS Filter Region: [all]
AWS Account: [REDACTED] UserId: [REDACTED]
Caller Identity ARN: [arn:aws:iam::REDACTED:user/secops]

Executing 247 checks, please wait...

**2023-02-08 13:21:36,659 [File: utils.py:15]     [Module: utils]  CRITICAL: /home/ubuntu/prowler/output/prowler-output-663596238219-20230208131920.json: OSError[13]**
-> Scanning sns service |▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▊⚠︎ | (!) 234/247 [95%] in 2:15.8 
(prowler) ubuntu@mkvmubuntu:~/prowler$

**Comment by toniblyx** [comment, marked as answer]
Hi @whatthejay, that is due to the file descriptor limits in your operating system, please see here to solve the problem https://docs.prowler.cloud/en/latest/troubleshooting/

**Reply by whatthejay** [reply]
Thanks @toniblyx , after modifying the value using 'ulimit -n 4096', it now works fine.
